Spec/WSL.g4

```diff
@@ -105,31 +104,16 @@ enumMember: Identifier ('=' constexpr)? ;
 funcDef: RESTRICTED? funcDecl block;
 funcDecl
     : (VERTEX | FRAGMENT) type Identifier parameters
-    | type (Identifier | OperatorName) typeParameters parameters
-    | OPERATOR typeParameters type parameters ;
+    | type (Identifier | OperatorName) parameters
+    | OPERATOR type parameters ;
 // Note: the return type is moved in a different place for operator casts, as a hint that it plays a role in overload resolution. to bikeshed
 parameters
     : '(' ')'
     | '(' parameter (',' parameter)* ')' ;
 parameter: type Identifier? ;
 
 nativeFuncDecl: RESTRICTED? NATIVE funcDecl ';' ;
-nativeTypeDecl: NATIVE TYPEDEF Identifier typeParameters ';' ;
-
-protocolDecl: PROTOCOL Identifier (':' protocolRef (',' protocolRef)*)? '{' (funcDecl ';')* '}' ;
-// Note: I forbid empty extensions lists in protocol declarations, while the original js parser allowed them. to bikeshed
-protocolRef: Identifier ;
-
-/*
- * Parser: Types 
- */
-typeParameters
-    : '<' typeParameter (',' typeParameter)* '>'
-    | ('<' '>')?;
-// Note: contrary to C++ for example, we allow '<>' and consider it equivalent to having no type parameters at all. to bikeshed
-typeParameter
-    : type Identifier
-    | Identifier (':' protocolRef ('+' protocolRef)*)? ;
+nativeTypeDecl: NATIVE TYPEDEF Identifier typeArguments ';' ;
 
 type
     : addressSpace Identifier typeArguments typeSuffixAbbreviated+
```

Comment:
Right. We’ll rely on the type checker to make sure typeArguments is the right thing.